### PR TITLE
add a solution to deploy argocd apps

### PR DESCRIPTION
**Pull request by elgnay**
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: solutions/deploy-argocd-apps/README.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    ```bash
+    export ADMIN_PASS=$(kubectl -n argocd get secret argocd-initial-admin-secret -o=jsonpath='{.data.password}' | base64 -d)
+    argocd login localhost:8080 --username=admin --password="${ADMIN_PASS}" --insecure
+    ```
+
+    Register the two OCM managed clusters to Argo CD
+
+    ```bash
+    argocd cluster add kind-cluster1 --name=cluster1
```
Comment (qiujian16) [inline solutions/deploy-argocd-apps/README.md:77]: will we be able to use a script to do this? or use managed-sa?

@mikeshng @xiangjingli when we have the ability to run argo with ManifestWork, this steps will be eased, right?

Comment (elgnay) [reply solutions/deploy-argocd-apps/README.md]: Done. A separated script , `setup-argocd.sh`, is create to integrate Argo CD with OCM.

Comment (xiangjingli) [reply solutions/deploy-argocd-apps/README.md]: @qiujian16   The solution is actually the push model of the argocd integration. Basically there is only one argocd server running on the hub cluster.  The argocd api server is exposed to localhost , then the argocd CLI is able to log into the argocd server, then uses argocd CLI to import the managed cluster secrets to the argocd api server NS, then create a applicationSet to directly deploy resources to the  remote managed clusters. 

In the new pull model of argocd integration we are working on, each managed cluster will run a local argocd  server. user needs to create a argocd application template on hub, and the argocd application template will be delivered to each managed cluster via manifestwork. Then the argocd server running on each managed cluster is triggered to deploy resources locally.  There is no need to import all managed cluster secrets to the ArgoCD api server. and we don't reply on argocd cli either.

With that being said, the push model and pull model don't conflict each other.  I think we can add another new solution to show how to deploy argocd app with pull model once it is available in upstream.

**Pull request merged**